Source/WebInspectorUI/UserInterface/Views/FontVariationDetailsSectionRow.js

```diff
@@ -142,6 +165,87 @@ WI.FontVariationDetailsSectionRow = class FontVariationDetailsSectionRow extends
 
         return 1;
     }
+
+    _showValidity()
+    {
+        if (!this._hasValidValue)
+            this.warningMessage = WI.UIString("Axis value outside of supported range: %s – %s", "A warning that is shown in the Font Details Sidebar when the value for a variation axis is outside of the supported range of values").format(this._formatAxisValueAsString(this._minimumValue), this._formatAxisValueAsString(this._maximumValue));
+        else
+            this.warningMessage = null;
+    }
+
+    _handleValueBlurEvent(event)
+    {
+        this._skipUpdatingInputValue = false;
+        this.value = this._value;
+    }
+
+    _handleValueInputEvent(event)
+    {
+        if (this._showValidityTimeout)
+            clearTimeout(this._showValidityTimeout);
+
+        // Allow the input to temporarily hold an invalid value without immediately warning the user.
+        this._showValidityTimeout = setTimeout(this._showValidity.bind(this), 750);
+
+        let valueAsString = event.target.value;
+        let valueAsNumber = event.target.valueAsNumber;
+
+        this._hasValidValue = false;
+        const validValueFormat = /\-?\d+(.\d+)?/g;
+
+        if (!validValueFormat.test(valueAsString)) {
+            this._skipUpdatingInputValue = true;
+
+            this.dispatchEventToListeners(WI.FontVariationDetailsSectionRow.Event.VariationValueChanged, {tag: this._tag, value: this._defaultValue});
+            return;
+        }
```

Comment:
Is there really no way we can prevent non-digits from being put in the field? Can we remove the the moment they are typed? Also, the warning message that would currently be used isn't entirely correct for the "this isn't even a number case". Saying the value is out of range undersells the real issue here. If we are going to allow non-digits to exist in this text field, we should have an error that tells you that only digits are allowed, or something to that effect...

Reply:
Actually, I see in `_handleValueKeydownEvent` we seem to prevent typing non-digits, so is this code ever executed?